engine/src/main/java/org/hibernate/validator/internal/engine/valueextraction/ValueExtractorManager.java

```diff
@@ -347,6 +348,104 @@ private static boolean isClassPresent(String className, boolean fallbackOnTCCL)
 		}
 	}
 
+	private final class ValueExtractorResolutionCache {
+
+		private final ConcurrentHashMap<ValueExtractorCacheKey, Set<ValueExtractorDescriptor>> possibleValueExtractorsByRuntimeTypes;
+		private final Set<Class<?>> nonContainerTypes;
+
+		public ValueExtractorResolutionCache() {
+			this.possibleValueExtractorsByRuntimeTypes = new ConcurrentHashMap<>();
+			this.nonContainerTypes = new HashSet<>();
```

Comment:
Let's initialize them directly when declaring the variable. There's no value doing it in the constructor (and we can remove the constructor).